collect_app/src/main/java/org/odk/collect/android/location/client/MapboxLocationClient.java

```diff
@@ -0,0 +1,33 @@
+package org.odk.collect.android.location.client;
+
+import android.location.Location;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.android.gms.location.LocationListener;
+import com.mapbox.android.core.location.LocationEngineCallback;
+import com.mapbox.android.core.location.LocationEngineResult;
+
+public class MapboxLocationClient implements LocationEngineCallback<LocationEngineResult> {
```

Comment:
I don't understand how the `LocationEngine` leak mentioned at https://docs.mapbox.com/android/core/guides/#requesting-location-updates would happen. Do you? Are you sure you're avoiding it here? It seems you still do have a strong reference to a Fragment here and so according to the example there it would be possible. It would be good to keep that link in the class comment since the structure is apparently important.

How about calling this `MapboxLocationCallback`? The `LocationClient` suffix to me implies it implements the interface of the same name. I think also augmenting the comment to say something like "Replace mock location accuracy with 0 as in `LocationClient` implementations since Mapbox uses its own location engine."

Reply:
Oh, `engine.requestLocationUpdates` holds a reference to an instance of this callback class, right? So if that's a strong reference and the reference to `locationListener` is also a strong reference I think that means neither the activity nor the engine will ever be garbage collected.

Reply:
Ok I added back WeakReference like it was in the original code. I think the issue is fixed now.